Dshop/apps/products_catalogue/tests/test_api_products.py

```diff
@@ -100,71 +98,122 @@ def assert_product_with_images(data):
 
 
 @pytest.mark.django_db
-def test_product_detail_with_images(authenticated_api_client, create_product_with_images):
+def test_product_detail_with_images(api_client_authed, create_product_with_images):
     url = reverse('products-api-detail', kwargs={'pk': create_product_with_images.id})
-    response = authenticated_api_client.get(url)
+    response = api_client_authed.get(url)
 
     assert response.status_code == 200
     assert_product_with_images(response.data)
 
 
 @pytest.mark.django_db
-def test_access_protected_resource(authenticated_api_client, create_active_product, create_inactive_product):
+def test_get_list_one(set_test_pagination_size, tv_product, inactive_product):
     url = reverse('products-api-list')
-    response = authenticated_api_client.get(url)
+    response = APIClient().get(url)
     assert response.status_code == 200
-
     results = response.data.get('results', [])
     assert len(results) == 1
-
     product_data = results[0]
-
+    assert response.data['count'] == 1
+    assert response.data['previous'] is None
+    assert response.data['next'] is None
     assert_active_object(product_data)
 
 
-def test_access_protected_resource_without_authentication(api_client):
-    url = reverse('products-api-list')
-    response = api_client.get(url)
+@pytest.mark.django_db
+def test_product_detail_404():
+    url = reverse('products-api-detail', kwargs={'pk': 6669})
+    response = APIClient().get(url)
+    assert response.status_code == 404
 
-    assert response.status_code == 401
+
+@pytest.fixture
+def set_test_pagination_size(settings, autouse=True):
+    settings.REST_FRAMEWORK['PAGE_SIZE'] = 5
 
 
+def test_pagination_size_in_tests():
```

Comment:
Please remove fixture since we moved to settings_tests.

Reply:
I want to know if settings_tests are changed. So we know the cause of test's failures